### PR TITLE
 fixes #5803 feat(reporting): Add celery task to generate reportlogs

### LINKED ISSUE #5803: [Reporting]: Add a celery tasks that generates reportinglogs
We need a celery task that generates reportlogs from experimentchangelogs, nimbuschangelogs and Normandy recipe history for easier aggregating of data

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-1428)

### PR DESCRIPTION
**Pull request by tiftran**
This commit:

    * adds a celery task that generates reportlogs from changelongs and normandy history

 Because:
    * This will help aggregate data for stats that will be generated

### CODE REVIEW

**Review by jaredlockhart — APPROVED** [review]
Looking great so far @tiftran  this is gonna be so useful 🙏

**Review by jaredlockhart — CHANGES_REQUESTED** [review]
Oh I don't think I meant to approve.

File: app/experimenter/reporting/tasks.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+        return get_update_event_reason(changelog)
+    return get_end_event_reason(changelog)
+
+
+def get_create_event_reason(changelog):
+    if changelog.message:
+        if "Cloned" in changelog.message:
+            return ReportLog.EventReason.CLONE
+        elif "Relaunch" in changelog.message:
+            return ReportLog.EventReason.RELAUNCH
```
[the remark this answers is not in the document]
Comment (tiftran) [reply app/experimenter/reporting/tasks.py]: hah, legacy has `experiment.parent`!,

File: app/experimenter/reporting/tasks.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+    for l_changelog in legacy_changelogs:
+        create_reportlog(l_changelog)
+
+    for n_changelog in nimbus_changelogs:
+        create_reportlog(n_changelog)
+
+    for experiment in live_experiments:
+        if experiment.normandy_id:
+            create_reportlog_from_normandy_history(experiment)
```
[the remark this answers is not in the document]
Comment (tiftran) [reply app/experimenter/reporting/tasks.py:36]: I tried it out locally, it didn't seem to have any problems, reportlogs were generated pretty quickly. Could we try the task without a cap? currently it's scheduled to run once a day?

[the remark this answers is not in the document]
Comment (jaredlockhart) [reply app/experimenter/reporting/tasks.py:36]: Yeah sounds good 👍 Also thanks for doing a test run!

File: app/experimenter/reporting/tasks.py [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+            experiment_name=experiment.name,
+            event_reason=ReportLog.EventReason.RECIPE_CHANGE,
+        ).exists()
+    return False
+
+
+def get_experiment_type(experiment):
+    if type(experiment) == NimbusExperiment:
+        application = experiment.application
+        application_slug = NimbusConstants.APPLICATION_CONFIGS[application].slug
```
Comment (jaredlockhart) [inline app/experimenter/reporting/tasks.py:115]: There's already a method for this:

https://github.com/mozilla/experimenter/blob/main/app/experimenter/experiments/models/nimbus.py#L225-L228

File: app/experimenter/reporting/tasks.py [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+        return get_end_event_reason(changelog)
+
+
+def get_create_event_reason(changelog):
+    experiment = changelog.experiment
+    if type(changelog) == ExperimentChangeLog:
+        if experiment.parent:
+            return ReportLog.EventReason.CLONE
+
+    return ReportLog.EventReason.NEW
```
Comment (jaredlockhart) [inline app/experimenter/reporting/tasks.py:164]: 👏 👏 👏 👏 👏 👏 👏

**Review by jaredlockhart — APPROVED** [review]
Alrighty!  Looking good!  There's a convenience method for looking up the application_config on a NimbusExperiment, but other than that looks pretty solid to me.  Great job @tiftran ! 🎉 🎉 🎉 🎉

File: app/experimenter/settings.py [hunk 5]
```diff
@@ -345,6 +345,10 @@
         "task": "experimenter.jetstream.tasks.fetch_jetstream_data",
         "schedule": 86400,
     },
+    "reporting_generate_report_logs": {
+        "task": "experimenter.reporting.tasks.generate_reportlogs",
+        "schedule": config("CELERY_REPORTING_INTERVAL", default=86400, cast=int),
```
Comment (jaredlockhart) [inline app/experimenter/settings.py:350]: Oh and you should add this setting to .env.sample

**Pull request merged**